### ISSUE TITLE
GameDig 5.x Support

### ISSUE THREAD

**Issue by scarecr0w12** [louislam/uptime-kuma#4414]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Gamedig 5.0.0-beta.0 Support

### ✔️ Solution

Please update package to include GameDig 5 for additional supported games, specifically PalWorld (one of the largest games to release on steam )

### ❓ Alternatives

None

**Comment by louislam** [comment]
Feel free to ping here again if they have released 5.0.0 non beta.

**Comment by scarecr0w12** [comment]
> Feel free to ping here again if they have released 5.0.0 non beta.

I'll be keeping an eye on it. As a work around, i should be able to change the package.json myself to include it locally, correct?

**Comment by louislam** [comment]
Yes, and run `npm install --production` to download the package.

**Comment by scarecr0w12** [comment]
hmm i guess the new version uses import function over require now, so code change would be needed.

**Comment by CommanderStorm** [comment]
Not the only breaking change:
> - Almost all games ids have been changed to follow a standard, see [CONTRIBUTING.md#naming](https://github.com/gamedig/node-gamedig/blob/5ae12dd494c927abcbe43352609d9aa34a54753c/CONTRIBUTING.md?plain=1#L27C3-L27C3).
> - Removed `minecraftping` (as it was deprecated and the same thing as `minecraft`) and `minecraftpe` (deprecated, which is now the same as `mbe` (Minecraft Bedrock Edition)).
> - Removed the `games.txt` file, the games definitions are now stored in-code (exported on games).

=> adding a migration with the renamings would be nessesary

That this also bumps the minimum node version to v16.
=> includes part of https://github.com/louislam/uptime-kuma/pull/3747

**Comment by FurkanVG** [comment]
Came here by to see that i would love this, for Palworld of course. Since there is no other way to monitor a Pal Server, i would live this.

**Comment by louislam** [comment]
> Not the only breaking change:
> 
> > * Almost all games ids have been changed to follow a standard, see [CONTRIBUTING.md#naming](https://github.com/gamedig/node-gamedig/blob/5ae12dd494c927abcbe43352609d9aa34a54753c/CONTRIBUTING.md?plain=1#L27C3-L27C3).
> > * Removed `minecraftping` (as it was deprecated and the same thing as `minecraft`) and `minecraftpe` (deprecated, which is now the same as `mbe` (Minecraft Bedrock Edition)).
> > * Removed the `games.txt` file, the games definitions are now stored in-code (exported on games).
> 
> => adding a migration with the renamings would be nessesary
> 
> That this also bumps the minimum node version to v16. => includes part of #3747

I think it is big enough to create a new monitor type for Gamedig 5.

**Comment by CommanderStorm** [comment]
Are you sure?

- I think that the parts needing migration (ids changing) can be easily migrated and the dropping of node v14 is not a big deal.
- `minecraftping` can be migrated to `minecraft`
- `minecraftpe` can be migrated to `mbe`
- Node 14 is EOL as of 19 Oct 2021 => a long time ago => we don't need to support that.

**Comment by chakflying** [comment]
I think for the end-user, it doesn't make sense to have another monitor type that basically does the same thing. But some work is definitely required to make the migration smooth.

**Comment by dustyhansen** [comment]
It looks like they released v 5.0 last month. Any more thought to implementing this so we can add some of the newer games?

**Comment by CommanderStorm** [comment]
If you are requesting guidance on how to implment this:
- please have a look at our [contribution guide](https://github.com/louislam/uptime-kuma/blob/bc25b719db25c5996c244a9cfad0ad20c3c766ed/CONTRIBUTING.md) how to setup your depelopment environmet
- This issue will require a migration to make the renamings and drop the games as noted in https://github.com/louislam/uptime-kuma/issues/4414#issuecomment-1907155718 
  How to write migrations is documented here: https://github.com/louislam/uptime-kuma/tree/master/db/knex_migrations
- Possibly, this issue requres to migrate the removal of the `games.txt` file as noted above


feel free to ping me if you have more questions ^^

**Comment by dustyhansen** [comment]
I was just letting you guys know that 5.0 was out of beta and hoping this would get some traction since gamedig hasn't been updated in 6 months+ and is missing a few games. I'm not a developer.

**Comment by elonmir** [comment]
/bump, missing Enshrouded... added in the 5.0.X Patch of Gamedig.

**Comment by elonmir** [comment]
Instead of just bumping I decided to do it myself... =)
